providers/multiprovider/src/main/java/dev/openfeature/contrib/providers/multiprovider/MultiProvider.java

```diff
@@ -0,0 +1,133 @@
+package dev.openfeature.contrib.providers.multiprovider;
+
+import dev.openfeature.sdk.EvaluationContext;
+import dev.openfeature.sdk.EventProvider;
+import dev.openfeature.sdk.FeatureProvider;
+import dev.openfeature.sdk.Metadata;
+import dev.openfeature.sdk.ProviderEvaluation;
+import dev.openfeature.sdk.Value;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.json.JSONObject;
+
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Provider implementation for Multi-provider.
+ */
+@Slf4j
+public class MultiProvider extends EventProvider {
+
+    @Getter
+    private static final String NAME = "multiprovider";
+    private final Map<String, FeatureProvider> providers;
+    private final Strategy strategy;
+    private String metadataName;
+
+    /**
+     * Constructs a MultiProvider with the given list of FeatureProviders, using a default strategy.
+     *
+     * @param providers the list of FeatureProviders to initialize the MultiProvider with
+     */
+    public MultiProvider(List<FeatureProvider> providers) {
+        this(providers, null);
+    }
+
+    /**
+     * Constructs a MultiProvider with the given list of FeatureProviders and a strategy.
+     *
+     * @param providers the list of FeatureProviders to initialize the MultiProvider with
+     * @param strategy the strategy
+     */
+    public MultiProvider(List<FeatureProvider> providers, Strategy strategy) {
+        this.providers = buildProviders(providers);
+        if (strategy != null) {
+            this.strategy = strategy;
+        } else {
+            this.strategy = new FirstMatchStrategy();
+        }
+    }
+
+    protected static Map<String, FeatureProvider> buildProviders(List<FeatureProvider> providers) {
+        Map<String, FeatureProvider> providersMap = new LinkedHashMap<>(providers.size());
+        for (FeatureProvider provider: providers) {
+            FeatureProvider prevProvider = providersMap.put(provider.getMetadata().getName(), provider);
+            if (prevProvider != null) {
+                log.warn("duplicated provider name: {}", provider.getMetadata().getName());
+            }
+        }
+        return Collections.unmodifiableMap(providersMap);
+    }
+
+    /**
+     * Initialize the provider.
+     * @param evaluationContext evaluation context
+     * @throws Exception on error
+     */
+    @Override
+    public void initialize(EvaluationContext evaluationContext) throws Exception {
+        JSONObject json = new JSONObject();
```

Comment:
Instead of JSON object, could we use a `HashMap`, or our SDK's own build-in object representation ([Structure](https://github.com/open-feature/java-sdk/blob/main/src/main/java/dev/openfeature/sdk/Structure.java) and it's implementations)?

Reply:
According to [Multi-provider metadata spec](https://github.com/open-feature/spec/blob/main/specification/appendix-a-included-utilities.md#metadata), it is expected to be printed as JSON, which makes sense.  
Both HashMap and Structure not printed as Json.